### PR TITLE
test: add api-connection e2e test cases

### PR DESCRIPTION
**Pull request by wenytang-ms**
E2E TEST: https://github.com/OfficeDev/TeamsFx/actions/runs/2165850262

### CODE REVIEW

File: packages/cli/tests/e2e/apiconnection/verifyScaffold.tests.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    await cleanUp(appName, projectPath, false, false, false);
+  });
+
+  it("scaffold with basic auth", async () => {
+    const basicInputs = "--user-name test123";
+    // action
+    await CliHelper.addApiConnection(projectPath, commonInputs, "basic", basicInputs);
+    // assert
+    const files = await fs.readdir(path.join(testFolder, "bot"));
+    console.log(files);
```
Comment (blackchoey) [inline packages/cli/tests/e2e/apiconnection/verifyScaffold.tests.ts:39]: remove the console.log here?

File: packages/cli/tests/e2e/apiconnection/verifyScaffold.tests.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    chai.expect(await fs.pathExists(path.join(testFolder, "bot", "test.js"))).to.be.true;
+    chai.expect(await fs.pathExists(path.join(testFolder, "bot", ".env.teamsfx.local"))).to.be.true;
```
Comment (blackchoey) [inline packages/cli/tests/e2e/apiconnection/verifyScaffold.tests.ts:41]: also check the content of .env file and package.json?

File: packages/cli/tests/e2e/apiconnection/verifyScaffold.tests.ts (hunk 1 above)
Comment (wenytang-ms) [reply packages/cli/tests/e2e/apiconnection/verifyScaffold.tests.ts:39]: still test the e2e case now

**Comment by wenytang-ms** [comment]
Full e2e tests: https://github.com/OfficeDev/TeamsFx/actions/runs/2165927273

**Review by blackchoey — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]

**Pull request merged**